Mixpanel/MPABTestDesignerConnection.m

```diff
@@ -255,21 +259,78 @@ - (void)webSocket:(MPWebSocket *)webSocket didCloseWithCode:(NSInteger)code reas
 
 - (void)showConnectedView
 {
-    if(!_recordingView) {
-        UIWindow *mainWindow = [[UIApplication sharedApplication] delegate].window;
-        _recordingView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, mainWindow.frame.size.width, 1.0)];
-        _recordingView.backgroundColor = [UIColor colorWithRed:4/255.0f green:180/255.0f blue:4/255.0f alpha:1.0];
-        [mainWindow addSubview:_recordingView];
-        [mainWindow bringSubviewToFront:_recordingView];
+    if (!self.connectivityIndicatorWindow) {
+        self.connectivityIndicatorWindow = [[UIWindow alloc] initWithFrame:CGRectMake(0, 20, CONNECTIVITY_INDICATOR_SIZE, CONNECTIVITY_INDICATOR_SIZE)];
+        self.connectivityIndicatorWindow.backgroundColor = [UIColor clearColor];
+        self.connectivityIndicatorWindow.windowLevel = UIWindowLevelAlert;
+        self.connectivityIndicatorWindow.alpha = 0;
+        [self.connectivityIndicatorWindow setHidden:NO];
+        UIImageView *bubbleView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, CONNECTIVITY_INDICATOR_SIZE, CONNECTIVITY_INDICATOR_SIZE)];
+        bubbleView.contentMode = UIViewContentModeScaleAspectFill;
+        bubbleView.image = [MPResources imageNamed:@"MPConnectivityIndicator"];
+        [self.connectivityIndicatorWindow addSubview:bubbleView];
+        
+        UIPanGestureRecognizer *panRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(moveConnectivityIndicator:)];
+        [self.connectivityIndicatorWindow addGestureRecognizer:panRecognizer];
+        indicatorAnimator = [[UIDynamicAnimator alloc] initWithReferenceView:self.connectivityIndicatorWindow];
+        
+        [UIView animateWithDuration:0.3 animations:^{
+            self.connectivityIndicatorWindow.alpha = 1;
+        }];
     }
 }
 
 - (void)hideConnectedView
 {
-    if (_recordingView) {
-        [_recordingView removeFromSuperview];
+    if (self.connectivityIndicatorWindow) {
+        [self.connectivityIndicatorWindow setHidden:YES];
+    }
+    self.connectivityIndicatorWindow = nil;
```

Comment:
Do we need to clean up the animator here? I image it keeps a strong reference to the window